keep/providers/openshift_provider/openshift_provider.py

```diff
@@ -13,17 +14,16 @@
 class OpenshiftProviderAuthConfig:
     """Openshift authentication configuration."""
 
-    api_server: str = dataclasses.field(
-        default=None,
+    api_server: pydantic.AnyHttpUrl = dataclasses.field(
         metadata={
             "name": "api_server",
             "description": "The openshift api server url",
             "required": True,
             "sensitive": False,
+            "validation": "any_http_url"
         },
```

Comment:
The idea behind which to use between `AnyHttpUrl` and `HttpsUrl` is basically based on whether the provider is open-source or can be self-hosted. This kind of providers can be run locally or on self-managed servers hence, the need for a url with an http scheme.

Red Hat offers self-managed editions of OpenShift, more information about this can be found [here](https://www.redhat.com/en/technologies/cloud-computing/openshift/self-managed)